lucene/core/src/java/org/apache/lucene/codecs/blocktree/FieldReader.java

```diff
@@ -91,7 +92,8 @@
       clone.seek(indexStartFP);
       // Initialize FST offheap if index is MMapDirectory and
       // docCount != sumDocFreq implying field is not primary key
-      if (clone instanceof ByteBufferIndexInput && this.docCount != this.sumDocFreq) {
+      isFSTOffHeap = clone instanceof ByteBufferIndexInput && ((this.docCount != this.sumDocFreq) || openedFromWriter == false);
```

Comment:
Is the idea here that it's only `IndexWriter` that needs fast ID lookups (since it uses this when deleting docs by ID field)?  E.g. apps that open `IndexReader` themselves (outside of `IndexWriter`) don't need fast ID lookups by default?

Reply:
I guess some users need fast lookups on read-only indices, but this proposal looks like a good default to me.

Reply:
@mikemccand you assumption is correct. @jpountz I will work on making it possible for users to  configure it down the road but I wanted to break this one out to make it simpler to iterate and not mix concerns.